engine/src/main/java/org/terasology/utilities/modifiable/ModifiableValue.java

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.utilities.modifiable;
+
+import gnu.trove.iterator.TFloatIterator;
+import gnu.trove.list.TFloatList;
+import gnu.trove.list.array.TFloatArrayList;
+
+/**
+ * A helper type to get and modify the value of a component without changing its actual value
+ * <p>
+ * The result value is guaranteed to be greater or equal to zero.
+ * Components using this type must mention so in their javadoc.
+ * </p>
+ */
+public class ModifiableValue {
```

Comment:
TODO: decide whether `ModifiableValue` should be mutable or immutable.

Reply:
Gone with the mutable approach in my latest commit #8876c27